src/sources/datadog/agent/tests.rs

```diff
@@ -902,3 +906,187 @@ async fn split_outputs() {
 
     COMPONENT_MULTIPLE_OUTPUTS_TESTS.assert(&["output"]);
 }
+
+#[test]
+#[allow(clippy::too_many_lines)]
+fn test_config_outputs() {
```

Comment:
I don't intend to add tests like these for all sources in the future, as we're basically copy/pasting the schema definition code of the codecs into this test, which isn't that useful, but I wanted to do it once for the first source, to make sure everything works as expected.

I might rewrite these tests in the future, to give them a more obvious place, but for now, this suffices to validate things working as expected, without overhauling too much w.r.t. testing.